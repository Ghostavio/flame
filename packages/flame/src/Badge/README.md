# Badge

Our basic badge component.

## Usage

First, make sure you have been through the [Getting started](https://github.com/lightspeed/flame#getting-started) steps of adding Flame in your application.

### React Components

#### `<Badge>`

| Prop       | Type                                                                     | Description                                                       |
| ---------- | ------------------------------------------------------------------------ | ----------------------------------------------------------------- |
| `children` | React.ReactNode                                                          | The content to display inside the button                          |
| `type`     | `'default', 'info', 'success', 'info', 'important', 'warning', 'danger'` | Style of badge                                                    |
| `size`     | `'small', 'medium'`                                                      | Change size of badge                                              |
| `bg`       | string                                                                   | Custom background color. Accepts any valid CSS color, i.e: `#000` |
| `color`    | string                                                                   | Custom text color. Accepts any valid CSS color, i.e: `#fff`       |

#### `<PillBadge>`

_extends `<Badge>`_

#### Example

```js
import React from 'react';
import { Badge, PillBadge } from '@lightspeed/flame/Badge';

const MyComponent = () => (
  <div>
    <Badge>My Badge</Badge>
    <PillBadge>My PillBadge</PillBadge>
  </div>
);

export default MyComponent;
```
