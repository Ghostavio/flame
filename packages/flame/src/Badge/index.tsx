import { Badge, BadgeProps, PillBadge } from './Badge';

export { Badge, BadgeProps, PillBadge };
