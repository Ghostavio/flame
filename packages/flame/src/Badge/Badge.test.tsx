/* eslint-disable import/no-extraneous-dependencies */
import * as React from 'react';

import { createComponent } from 'test-utils';
import { Badge, PillBadge } from './Badge';

const types = ['default', 'success', 'info', 'warning', 'danger', 'important'];

describe('<Badge />', () => {
  it('renders correctly', () => {
    const component = createComponent(<Badge>Hello World</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders small variant', () => {
    const component = createComponent(<Badge size="small">Hello World</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders pill variant', () => {
    const component = createComponent(<PillBadge>Hello World</PillBadge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders with custom background and text color', () => {
    const component = createComponent(
      <Badge bg="#ddd" color="#222">
        Hello World
      </Badge>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders with custom background only', () => {
    const component = createComponent(<Badge bg="#ddd">Hello World</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders with custom text color only', () => {
    const component = createComponent(<Badge color="#222">Hello World</Badge>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  types.forEach(type => {
    it(`renders correctly with ${type} type`, () => {
      const component = createComponent(<Badge type={type as any}>Hello World</Badge>);
      expect(component.toJSON()).toMatchSnapshot();
    });
  });
});
