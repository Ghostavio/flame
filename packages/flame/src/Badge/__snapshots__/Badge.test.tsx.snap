// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> renders correctly 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #abb3b3;
  color: #0c0d0d;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="default"
>
  Hello World
</span>
`;

exports[`<Badge /> renders correctly with danger type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #b93435;
  color: #fff;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="danger"
>
  Hello World
</span>
`;

exports[`<Badge /> renders correctly with default type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #abb3b3;
  color: #0c0d0d;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="default"
>
  Hello World
</span>
`;

exports[`<Badge /> renders correctly with important type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #2875c6;
  color: #fff;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="important"
>
  Hello World
</span>
`;

exports[`<Badge /> renders correctly with info type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #5187e0;
  color: #fff;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="info"
>
  Hello World
</span>
`;

exports[`<Badge /> renders correctly with success type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #35b858;
  color: #fff;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="success"
>
  Hello World
</span>
`;

exports[`<Badge /> renders correctly with warning type 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #f39c26;
  color: #494c4c;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="warning"
>
  Hello World
</span>
`;

exports[`<Badge /> renders pill variant 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #abb3b3;
  color: #0c0d0d;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
  padding-left: 0.75rem;
  padding-right: 0.75rem;
  border-radius: 10rem;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="default"
>
  Hello World
</span>
`;

exports[`<Badge /> renders small variant 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #abb3b3;
  color: #0c0d0d;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.5rem;
  line-height: 1.125rem;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
}

<span
  className="emotion-0 emotion-1"
  size="small"
  type="default"
>
  Hello World
</span>
`;

exports[`<Badge /> renders with custom background and text color 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #abb3b3;
  color: #0c0d0d;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
  background-color: #ddd;
  color: #222;
}

<span
  className="emotion-0 emotion-1"
  color="#222"
  size="medium"
  type="default"
>
  Hello World
</span>
`;

exports[`<Badge /> renders with custom background only 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #abb3b3;
  color: #0c0d0d;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
  background-color: #ddd;
}

<span
  className="emotion-0 emotion-1"
  size="medium"
  type="default"
>
  Hello World
</span>
`;

exports[`<Badge /> renders with custom text color only 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-transform: uppercase;
  font-weight: bold;
  border-radius: 0.1875rem;
  background-color: #abb3b3;
  color: #0c0d0d;
  padding-left: 0.375rem;
  padding-right: 0.375rem;
  font-size: 0.75rem;
  line-height: 1.5rem;
  -webkit-letter-spacing: 0.09375rem;
  -moz-letter-spacing: 0.09375rem;
  -ms-letter-spacing: 0.09375rem;
  letter-spacing: 0.09375rem;
  color: #222;
}

<span
  className="emotion-0 emotion-1"
  color="#222"
  size="medium"
  type="default"
>
  Hello World
</span>
`;
